### PR TITLE
feat: add --output json global flag for list/detail/whoami

### LINKED ISSUE #140: feat(all): add `--output json` flag to all commands for machine-readable output
### Background

AI coding agents (Claude Code, Codex CLI, Cursor, etc.) and CI/CD pipelines are becoming primary users of the Alpacon CLI. Currently, `list` commands output only in human-readable table format, and `detail/update` commands output only in pretty-printed JSON. Parsing table output is fragile for agents, and even pretty-printed JSON can vary across tooling.

The `whoami --json` command already establishes this pattern, so we extend it to all output commands.

### Current state

| Output format | Command count | Examples |
|---|---|---|
| `PrintTable` (table only) | 18 | `server ls`, `user ls`, `cert ls`, `event ls`, etc. |
| `PrintJson` (pretty JSON only) | 15 | `server desc`, `user desc`, `cert desc`, etc. |
| `--json` flag supported | 1 | `whoami` |

### Proposal

Add a global `--output json` flag (short: `-o json`) to all output commands.

**`PrintTable` commands (18):**

Default is table; `--output json` outputs a JSON array.

```
$ alpacon server ls
NAME        STATUS    OS
my-server   online    Ubuntu 24.04

$ alpacon server ls --output json
[{"name":"my-server","status":"online","os":"Ubuntu 24.04", ...}]
```

**`PrintJson` commands (15):**

Default is pretty JSON (unchanged); `--output json` outputs compact JSON.

### Target commands

**PrintTable (list commands):**
- `server ls`, `user ls`, `group ls`, `token ls`, `token acl ls`
- `cert ls`, `csr ls`, `authority ls`, `note ls`
- `package system ls`, `package python ls`
- `event ls`, `log ls`, `audit ls`
- `websh ls`, `webftp ls`, `webhook ls`
- `workspace ls`, `revoke ls`

**PrintJson (detail/update commands):**
- `server desc`, `server update`, `user desc`, `user update`
- `group desc`, `group update`, `cert desc`, `csr desc`
- `authority desc`, `authority update`
- `note desc`, `note update`
- `webhook desc`, `webhook update`
- `websh desc`, `revoke desc`

### Implementation approach

1. Add `--output` persistent flag to `RootCmd` (`table` | `json`, default `table`)
2. Add an output helper in `utils` that branches on the flag (table vs JSON)
3. Replace `PrintTable`/`PrintJson` calls in each command with the new helper
4. Keep `whoami --json` backward-compatible; `--output json` also works

### Expected impact

- AI agents can reliably parse CLI output
- CI/CD scripts can post-process with `jq` and similar tools
- No change to existing user experience (default remains current behavior)

### PR DESCRIPTION
**Pull request by jisung-02**
## Summary

Closes #140.

Introduces a `--output` persistent flag on `RootCmd` (`table` | `json`, default `table`). When `--output json` is set, `utils.PrintTable()` emits a JSON array and `utils.PrintJson()` re-indents response bodies, both with 2-space pretty-printing suitable for AI agents, CI/CD, and `jq` pipelines. Function signatures are unchanged — the format is read from a package global bound to the flag — so all ~47 existing call sites remain untouched. `whoami`'s deprecated `--json` boolean is removed in favor of the unified convention.

No short form is registered — `-o` is reserved for subcommand-local `--out` flags (e.g., `cert download -o path`).

## Changes

- \`utils/output.go\` — add \`OutputFormat\` package global + \`OutputFormatTable\`/\`OutputFormatJSON\` constants; insert JSON branch in \`PrintTable\` (empty/nil slice → \`[]\`) and \`PrintJson\`
- \`cmd/root.go\` — register \`--output\` persistent flag via \`StringVar(&utils.OutputFormat, ...)\`; add \`PersistentPreRunE\` validator rejecting invalid values
- \`cmd/whoami.go\` — drop \`--json\` flag and its \`init()\` registration; route through \`utils.OutputFormat\` and \`utils.PrintJson\`
- \`utils/output_test.go\` — new; 6 tests covering JSON/table branches of both helpers, including empty/nil slice edge cases
- \`CLAUDE.md\` — document the new flag in Key patterns

## Behavior

\`\`\`
alpacon server list                        # table (unchanged)
alpacon server list --output json          # pretty JSON array
alpacon server describe <id> --output json # pretty JSON object
alpacon whoami --output json               # pretty JSON object
alpacon server list --output bogus         # errors: invalid --output value
\`\`\`

Empty/nil list results emit \`[]\n\` so \`jq\` pipelines don't break on no-data.

## Breaking changes

- \`alpacon whoami --json\` flag removed. Replacement: \`alpacon whoami --output json\`
- \`whoami --output json\` output is now pretty-printed with 2-space indent (previously \`--json\` produced 2-space pretty; behavior now consistent with the rest of the CLI)

## Checklist

- [x] Code follows project style guidelines
- [x] Self-reviewed the code
- [x] No \`fmt.Println\` debug statements left
- [x] No hardcoded secrets or credentials
- [x] Updated relevant documentation (CLAUDE.md)
- [x] Error handling via \`utils.CliErrorWithExit\` (errcheck clean)
- [x] Tests cover edge cases (empty slice, nil slice, both format branches)
- [x] \`go build\`, \`go vet\`, \`go test -race ./...\` all pass
- [x] No new dependencies (\`go mod tidy\` clean)

## Test plan

- [ ] \`alpacon server list\` — table (regression check)
- [ ] \`alpacon server list --output json | jq .\` — parses
- [ ] \`alpacon server describe <id> --output json\` — pretty JSON
- [ ] \`alpacon whoami --output json | jq .username\` — parses
- [ ] \`alpacon whoami --json\` — errors with unknown flag (expected)
- [ ] \`alpacon server list --output bogus\` — errors with invalid value (expected)
- [ ] Empty list result → \`[]\`

### CODE REVIEW

File: cmd/whoami.go [hunk 1]
```diff
@@ -36,9 +36,9 @@ var whoamiCmd = &cobra.Command{
 and permissions. Useful for verifying context before running infrastructure
 commands, especially for AI agents and operators managing multiple workspaces.`,
 	Example: `  alpacon whoami
-  alpacon whoami --json`,
+  alpacon whoami --output json`,
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/whoami.go:39]: The `Example` string content is already consistent — both lines share the same 2-space indent. The visual offset in the source file comes from the Go string literal layout (line 38 starts after `\tExample: \``, line 39 starts at column 0), but the runtime string `--help` prints is:

```
  alpacon whoami
  alpacon whoami --output json
```

Verified with `alpacon whoami --help`. No change needed.

File: cmd/root.go [hunk 2]
```diff
@@ -51,6 +51,12 @@ func Execute() {
 }
 
 func init() {
+	// Global output format flag
+	RootCmd.PersistentFlags().StringVarP(
+		&utils.OutputFormat, "output", "o", "table",
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/root.go]: Good catch — the conflict was real. Six subcommands (`cmd/cert/cert_download.go`, `cmd/csr/csr_download_crt.go`, `cmd/csr/csr_create.go`, `cmd/authority/authority_download.go`, `cmd/authority/authority_download_crl.go`) all bind `-o` to a local `--out` flag for file paths.

Dropped the `-o` shorthand in 5aaf129. `--output` is long-form only now, and CLAUDE.md reserves `-o` for subcommand-local `--out` flags.

File: cmd/root.go [hunk 3]
```diff
@@ -51,6 +51,12 @@ func Execute() {
...
+		"Output format: table or json",
+	)
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/root.go:67]: Agreed — silent fallback to table output on typos like `--output jsno` is a poor UX for AI agents and CI scripts, which is exactly the audience this flag was added for.

Fixed in d8f61ae: added a `PersistentPreRunE` on `RootCmd` that rejects anything other than `table`/`json` with a clear error message. Also extracted `utils.OutputFormatTable` / `utils.OutputFormatJSON` constants so the accepted values live in one place.

File: utils/output.go [hunk 4]
```diff
@@ -76,6 +93,15 @@ func PrintTable(slice any) {
 }
 
 func PrintJson(body []byte) {
+	if OutputFormat == "json" {
+		var buf bytes.Buffer
+		if err := json.Indent(&buf, body, "", "  "); err != nil {
+			CliErrorWithExit("Parsing data: Expected a json format")
+		}
+		fmt.Fprintln(os.Stdout, buf.String())
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply utils/output.go]: Fixed in 7908dd8 — both error messages now read `"Parsing data: Expected a JSON format."` (capitalized acronym + trailing period) in both the `--output json` branch and the default pretty-print branch.

File: utils/output_test.go [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+	go func() {
+		var buf bytes.Buffer
+		_, _ = io.Copy(&buf, r)
+		done <- buf.String()
+	}()
+
+	fn()
+	_ = w.Close()
+	os.Stdout = old
+	return <-done
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply utils/output_test.go:43]: Fixed in 693419c — registered a `t.Cleanup` that restores `os.Stdout` and closes both pipe ends, so a panic in `fn()` can no longer leave `os.Stdout` bound to a dead pipe or leak the `io.Copy` goroutine / file descriptors.

The explicit close/restore on the happy path is kept because `w.Close()` must run before `<-done` to unblock `io.Copy`; the Cleanup is a safety net and the resulting double-close on the happy path is harmless.

File: cmd/root.go [hunk 6]
```diff
@@ -51,6 +60,12 @@ func Execute() {
 }
 
 func init() {
+	// Global output format flag
+	RootCmd.PersistentFlags().StringVar(
+		&utils.OutputFormat, "output", utils.OutputFormatTable,
+		"Output format: table or json",
+	)
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/root.go:67]: Fixed — updated the PR description to drop every `-o json` reference (Summary, Behavior block, Test plan) and clarified that no short form is registered because `-o` is reserved for subcommand-local `--out` flags. The actual flag surface is now `--output` only.

File: cmd/root.go [hunk 7]
```diff
@@ -38,6 +38,15 @@ tunnels, and manage certificates — all with zero-trust authentication, MFA,
...
 Designed to be used by engineers, AI coding agents, and CI/CD platforms alike.`,
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		switch utils.OutputFormat {
+		case utils.OutputFormatTable, utils.OutputFormatJSON:
+			return nil
+		default:
+			return fmt.Errorf("invalid --output value %q (expected %q or %q)",
+				utils.OutputFormat, utils.OutputFormatTable, utils.OutputFormatJSON)
+		}
+	},
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/root.go:49]: Acknowledged, but intentionally not fixing in this PR. Neither `exec` nor `websh` calls `utils.PrintTable` or `utils.PrintJson` — their stdout is the remote command output / terminal stream, so `--output` is structurally meaningless for them. The only user-visible effect today is a confusing error message (exec treats `--output` as part of the remote command; websh treats it as the server name), not a behavioral regression. Stripping or explicitly rejecting `--output` inside those two manual parsers is a follow-up UX polish that belongs with a dedicated review of the `DisableFlagParsing` commands, not with the list/detail/whoami output-format rollout. Tracking separately.

File: utils/output.go [hunk 8]
```diff
@@ -14,13 +14,36 @@ import (
...
+		if s.IsNil() || s.Len() == 0 {
+			_, _ = fmt.Fprintln(os.Stdout, "[]")
+			return
+		}
+		data, err := json.MarshalIndent(slice, "", "  ")
+		if err != nil {
+			CliErrorWithExit("Failed to marshal data to JSON: %s", err)
+		}
+		_, _ = fmt.Fprintln(os.Stdout, string(data))
+		return
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply utils/output.go:44]: Fixed in c3b66e3. A sweep across every type passed to `utils.PrintTable` found two offenders — `api/websh.SessionListItem` and `api/workspace.WorkspaceListEntry` (the latter had no struct tags at all). Both now carry snake_case `json` tags so `--output json` is stable across all list commands. Every other PrintTable target (cert, iam, server, audit, webhook, auth, event, webftp, note, packages, security) already had matching `json` tags, so no further changes were needed.

**Review by eunyoung14 — APPROVED** [review]
